audeer/core/io.py

```diff
@@ -106,21 +107,88 @@ def create_archive(
 
     Args:
         root: path to root folder of archive.
-            Path names inside the archive
-            will be relative to ``root``
-        files: files to include in archive,
-            relative to ``root``
+            Only files below ``root`` can be included
+            and will be stored relative to ``root``
+        files: files that will be included in the archive.
+            Absolute and relative file paths are possible,
+            as long as the files are below ``root``.
+            If set to ``None``
+            all files below ``root``
+            will be added to the archive.
+            This includes hidden files
+            and files from sub-folders
         archive: path to archive file.
-            The archive type is determined by the file extension
+            The type of the archive
+            is determined from its file extension
         verbose: if ``True`` a progress bar is shown
 
     Raises:
+        FileNotFoundError: if ``root`` or a file in ``files`` is not found
+        NotADirectoryError: if ``root`` is not a directory
         RuntimeError: if archive does not end with ``zip`` or ``tar.gz``
+        ValueError: if ``file`` is not below ``root``
 
     """
+    root = safe_path(root)
     archive = safe_path(archive)
     mkdir(os.path.dirname(archive))
-    files = to_list(files)
+
+    if not os.path.exists(root):
+        raise FileNotFoundError(
+            errno.ENOENT,
+            os.strerror(errno.ENOENT),
+            root,
+        )
+
+    if not os.path.isdir(root):
+        raise NotADirectoryError(
+            errno.ENOTDIR,
+            os.strerror(errno.ENOTDIR),
+            root,
+        )
+
+    if files is None:
+
+        files = list_file_names(
+            root,
+            basenames=True,
+            recursive=True,
+            hidden=True,
+        )
+
+    else:
+
+        files_org = to_list(files)
+        files = []
+
+        for file in files_org:
+
+            # convert to absolute path
+            if not os.path.isabs(file):
+                path = safe_path(root, file)
+            else:
+                path = safe_path(file)
+
+            # file is below root
+            if not path.startswith(root):
+                raise ValueError(
```

Comment:
If archive does not end on ZIP or TAR.GZ we raise a `RuntimeError`. Should we raise a `RuntimeError` for consistency here, too?

Reply:
Yes, I would change to `RuntimeError`.

Reply:
done